_posts/is-it-complex-break-it-down.md

```diff
@@ -0,0 +1,89 @@
+
+## Next, please
+
+> Once converted, the process proceeds to fulfill and prepare the order for shipment.
+
+If a service in the system already broadcasts an event to notify that an order has been created, warehouse, another service, can react and proceed with orders fulfillment. Similarly, can shipping.
+
+From a service boundaries perspective, fulfillment and shipping are not part of the order management "thing" because the order management thing does not exist. Order management is a "user mental model" concept that doesn't translate into a service.
+
+Fulfillment and shipping are long-running business transactions, though. They are both started by the order-created event. Fulfillment, during its lifecycle, will collect items from warehouse shelves and take care of all the exceptions connected to no more in-stock items. It'll do so by raising events, not directly managing the exception. Other services will handle those events and make decisions on how to proceed. One of these might be shipping that can turn what was supposed to be a single shipment into multiple deliveries once stocks are replenished.
+
+### Shipping is a long-running business process too
+
+Similar to fulfillment, shipping starts with the order-created event and reacts to interesting events broadcasted by other services. For example, a fulfillment-completed event might trigger the final steps to deliver the order to the customer.
+
+> When the order is shipped, an invoice is issued and marked as due based on its payment terms. Later on, the order management system checks for overdue invoices.
+
+All that said applies to finance too. Finance reacts to the order-shipped event, broadcasted by shipping, and issues the required invoice. Invoicing implies charging the user's credit card, which requires authorizing the card first.
+
+### Surprise, surprise, finance is a long-running process too
+
+We can create a separate saga, or more than one, to handle the payment and invoicing procedures. Payment is a different saga than the one needed to generate the invoice. But it doesn't end here. Once the invoice is issued, which turns to be an event published by finance, a third saga is kicked off to keep track of the shipment-returned event to refund the user's credit card eventually. Another option could be if the system allows different payment methods: if customers could buy and pay, upon receiving the goods, using a wire transfer, the system needs to keep track of overdue payments.
+
+> A sample implementation of this last bit is described in [Got the time article](https://milestone.topics.it/2021/03/05/got-the-time.html).
```

Comment:
If you're feeling ambitious, consider expanding on the sample to incorporate all (or most) of the concepts you mention above.